src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordGetHashCode.cs

```diff
@@ -49,10 +49,15 @@ internal override void GenerateMethodBody(TypeCompilationState compilationState,
             {
                 MethodSymbol? equalityComparer_GetHashCode = null;
                 MethodSymbol? equalityComparer_get_Default = null;
-                BoundExpression currentHashValue;
+                BoundExpression? currentHashValue;
 
-                if (ContainingType.BaseTypeNoUseSiteDiagnostics.IsObjectType())
+                if (ContainingType.IsStructType())
                 {
+                    currentHashValue = null;
```

Comment:
>currentHashValue [](start = 20, length = 16)

Should we add hash code for the type in order to add variance across different types of record structs?

Reply:
I'm not sure I follow. You mean if we did struct inheritance?

---
In reply to: [597084774](https://github.com/dotnet/roslyn/pull/51900#discussion_r597084774) [](ancestors = 597084774)

Reply:
> I'm not sure I follow. You mean if we did struct inheritance?

No, inheritance isn't part of this. This is about instances of ```record struct S1;``` and instances of ```record struct S2;``` having distict hash codes. The same way we do this for record classes. 

---
In reply to: [597882249](https://github.com/dotnet/roslyn/pull/51900#discussion_r597882249) [](ancestors = 597882249,597084774)

Reply:
Current spec doesn't include that.
I'll add an open question: [https://github.com/dotnet/csharplang/pull/4564](https://github.com/dotnet/csharplang/pull/4564)

---
In reply to: [597887909](https://github.com/dotnet/roslyn/pull/51900#discussion_r597887909) [](ancestors = 597887909,597882249,597084774)